src/lib.rs

```diff
@@ -272,6 +275,33 @@ impl StreamDeck {
         self.write_button_image(key, &image)
     }
 
+    /// Sets a button to the provided text.
+    /// Will break text over \n linebreaks
+    pub fn set_button_text(
+        &mut self,
+        key: u8,
+        colour: &Colour,
+        background: &Colour,
+        x: u32,
+        y: u32,
+        scale: Scale,
```

Comment:
would it be reasonable to pack the less-used options here into a `TextOptions` object with a `Default` impl? i'm not sure where exactly the line is, but, anything shared between calls to set sequential buttons (or the same button in different states) might be a useful guideline.

not intended to be prescriptive in any way, but something along the lines of `set_button_test(&mut self, text: &str, fg: &Colour, bg: &Colour, options: &TextOptions)` that elides any customisation that isn't _strictly_ required through `TextOptions::default()`.

Reply:
Ooh, that's an interesting idea. Let me have think, and will update the PR.